packages/wonder-blocks-modal/components/modal-launcher/modal-launcher-portal.js

```diff
@@ -0,0 +1,80 @@
+// @flow
+/**
+ * An element used by ModalLauncher, that behaves similarly to a React
+ * "portal". It renders its children in a new node in `document.body`, rather
+ * than in its natural position in the DOM tree.
+ *
+ * NOTE(mdr): Once we upgrade to React 16, maybe we can use built-in portals!
+ */
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import {View} from "wonder-blocks-core";
+
+type Props = {
+    children: React.Node,
+};
+
+export default class ModalLauncherPortal extends React.Component<Props> {
+    _destination: ?Node;
+
+    /**
+     * When we mount, we also mount our children into a new child of
+     * `document.body`.
+     */
+    componentDidMount() {
+        const root = this._getRoot();
+
+        // Create a new destination node, and add it to the root.
+        // The data attribute is used in unit tests, to identify which
+        // ancestors of `children` were created by `ModalLauncherPortal`.
+        const destination = document.createElement("div");
+        destination.setAttribute("data-modal-launcher-portal", "");
+        root.appendChild(destination);
+
+        // We'll also wrap the children in a wrapper node, so that we can
+        // accept `React.Node` instead of `React.Element`.
+        //
+        // NOTE(mdr): Once we upgrade to React 16, we'll no longer need this
+        //     wrapper node!
```

Comment:
The whole `ModalLauncherPortal` component can go away with React 16, right?

Reply:
Not quite, unfortunately. Built-in portals enable us to use an existing node as our render root, but don't handle creating/removing the node for you. So, we'd still have to manage the destination node ourselves.

The difference with built-in portals would be that, instead of using `ReactDOM` methods in mount/unmount, we could return a `ReactDOM.createPortal()` from `render` instead.